test/integration_test.js

```diff
@@ -237,53 +425,191 @@ describe("Integration tests", () => {
+          });
+
+          it("should not skip records", () => {
+            expect(syncResult.skipped).to.have.length.of(0);
+          });
+
+          it("should not import anything", () => {
+            expect(syncResult.created).to.have.length.of(0);
+          });
+
+          it("should not publish anything", () => {
+            expect(syncResult.published).to.have.length.of(0);
+          });
+
+          it("should not update anything", () => {
+            expect(syncResult.updated).to.have.length.of(0);
+          });
+
+          it("should list resolved records", () => {
+            expect(syncResult.resolved).to.have.length.of(1);
+            expect(syncResult.resolved[0].title).eql("task1-local");
+          });
+
+          it("should put local database in the expected state", () => {
+            return tasks.list({order: "title"})
+              .then(res => res.data.map(record => ({
+                title: record.title,
+                _status: record._status,
+              })))
+              .should.become([
+                // For CLIENT_WINS strategy, local version is marked as synced
+                {title: "task1-local", _status: "synced"},
+              ]);
+          });
         });
 
-        it("should not have updated anything", () => {
-          expect(syncResult.updated).to.have.length.of(0);
+        describe("SERVER_WINS strategy", () => {
+          beforeEach(() => {
+            return tasks.sync({strategy: Kinto.syncStrategy.SERVER_WINS})
+            .then(res => {
+              syncResult = res;
+            });
+          });
+
+          it("should not have an ok status", () => {
+            expect(syncResult.ok).eql(true);
+          });
+
+          it("should contain no errors", () => {
+            expect(syncResult.errors).to.have.length.of(0);
+          });
+
+          it("should have a valid lastModified value", () => {
+            expect(syncResult.lastModified).to.be.a("number");
+          });
+
+          it("should have the outgoing conflict listed", () => {
+            expect(syncResult.conflicts).to.have.length.of(0);
+          });
+
+          it("should not skip records", () => {
+            expect(syncResult.skipped).to.have.length.of(0);
+          });
+
+          it("should not import anything", () => {
+            expect(syncResult.created).to.have.length.of(0);
+          });
+
+          it("should not publish anything", () => {
+            expect(syncResult.published).to.have.length.of(0);
+          });
+
+          it("should not update anything", () => {
+            expect(syncResult.updated).to.have.length.of(0);
+          });
+
+          it("should list resolved records", () => {
+            expect(syncResult.resolved).to.have.length.of(1);
+            expect(syncResult.resolved[0].title).eql("task1-remote");
+          });
+
+          it("should put local database in the expected state", () => {
```

Comment:
ditto: add a test to make sure nothing is send to the server which is already up-to-date.